tests/cpp/core/LoggingUtilityTest.cpp

```diff
@@ -0,0 +1,199 @@
+
+    QObject::connect( displayGroup.get(), &DisplayGroup::contentWindowAdded,
+                      logger.get(), &LoggingUtility::contentWindowAdded );
+
+    displayGroup->addContentWindow( window1 );
+    BOOST_CHECK( logger.get()->getWindowCount( ) == 1) ;
+    
+    displayGroup->removeContentWindow( window1 );
+    BOOST_CHECK( logger.get()->getWindowCount( ) == 1) ;
+
+}
+
+BOOST_AUTO_TEST_CASE( testRecentInteractionCounter )
+{
+    ContentPtr content( new DummyContent );
+    DisplayGroupPtr displayGroup( new DisplayGroup( wallSize ));
+
+    ContentWindowPtr window1 = boost::make_shared<ContentWindow>( content );
+    ContentWindowPtr window2 = boost::make_shared<ContentWindow>( content );
+    std::unique_ptr<LoggingUtility> logger = make_unique<LoggingUtility>();
+
+    BOOST_CHECK( logger.get()->getRecentInteractionCounter( ) == 0) ;
+
+    QObject::connect( displayGroup.get(), &DisplayGroup::contentWindowAdded,
+                      logger.get(), &LoggingUtility::contentWindowAdded );
+ 
+    displayGroup->addContentWindow( window1 );
+    displayGroup->focus( window1->getID() );
+    BOOST_CHECK( logger.get()->getRecentInteractionCounter( ) == 2) ;
+
+}
+
+BOOST_AUTO_TEST_CASE( testLastInteraction )
+{
+    ContentPtr content( new DummyContent );
+    DisplayGroupPtr displayGroup( new DisplayGroup( wallSize ));
+    
+    ContentWindowPtr window1 = boost::make_shared<ContentWindow>( content );
+    std::unique_ptr<LoggingUtility> logger = make_unique<LoggingUtility>();
+    BOOST_CHECK( logger.get()->getLastInteraction( ) == "") ;
+
+    QObject::connect( displayGroup.get(), &DisplayGroup::contentWindowAdded,
+                      logger.get(), &LoggingUtility::contentWindowAdded );
+ 
+    displayGroup->addContentWindow( window1 );
+    displayGroup->focus( window1->getID() );
+    BOOST_CHECK( logger.get()->getLastInteraction( ) == "mode changed") ;
+
+}
+
+BOOST_AUTO_TEST_CASE( testJsonOutput )
+{
+    ContentPtr content( new DummyContent );
+    DisplayGroupPtr displayGroup( new DisplayGroup( wallSize ));
+    
+    ContentWindowPtr window1 = boost::make_shared<ContentWindow>( content );
+    ContentWindowPtr window2 = boost::make_shared<ContentWindow>( content );
+    std::unique_ptr<LoggingUtility> logger = make_unique<LoggingUtility>();
+
+    QObject::connect( displayGroup.get(), &DisplayGroup::contentWindowAdded,
+                      logger.get(), &LoggingUtility::contentWindowAdded );
+ 
+    std::unique_ptr<RestLogger> logContent;
+    logContent.reset( new RestLogger( *(logger.get()) ));
+    BOOST_CHECK( logContent->_toJSON() == defaultJson ) ;
+
+    displayGroup->addContentWindow( window1 );
+    displayGroup->addContentWindow( window2 );
+
+    std::string logText =  logContent->_toJSON() ;
+ 
+    boost::regex ip_regex(regexJson);
+    boost::sregex_iterator it(logText.begin(), logText.end(), ip_regex);
+    boost::sregex_iterator end;
+
+    std::string outputJson;
+    for (; it != end; ++it)
+    {
```

Comment:
Style note: we generally omit curly braces when the body of a for-loop is only one line